src/components/popups-etc/chat-component.tsx

```diff
@@ -88,7 +90,12 @@ Answer the questions that follow based on this unless new code is provided.`;
 			input.value = "";
 
 			// send code as message to give context to the llm for future questions
-			await sendMessage(newSystemMessage.content);
+			// send new code only if the code has changed since the last one
+			const newCodeHash = await sha256Hash(codeMirror.value?.state.doc.toString()!);
+			if (newCodeHash !== codeHash) {
+			    setCodeHash(newCodeHash);
+					await sendMessage(newSystemMessage.content);
```

Comment:
do we fail gracefully if this fails...?

Reply:
it'll only set the code hash if the request is successful. 
there error will also be bubbled-up and handle in the top-level try-catch